src/applications/disability-benefits/all-claims/pages/toxicExposure/toxicExposureChoicePage.jsx

```diff
@@ -0,0 +1,358 @@
+  ] = useState(false);
+  const [
+    showDeletedToxicExposureConfirmation,
+    setShowDeletedToxicExposureConfirmation,
+  ] = useState(false);
+
+  const deletedToxicExposureConfirmationRef = useRef(null);
+
+  useEffect(
+    () => {
+      if (
+        showDeletedToxicExposureConfirmation &&
+        deletedToxicExposureConfirmationRef.current
+      ) {
+        deletedToxicExposureConfirmationRef.current.focus();
+      }
+    },
+    [showDeletedToxicExposureConfirmation],
+  );
+
+  const modalRef = useRef(null);
+  useEffect(
+    () => {
+      if (showDeleteToxicExposureModal && modalRef.current) {
+        const modalHeading = document.querySelector('h4');
+        scrollAndFocus(modalHeading);
+      }
+    },
+    [showDeleteToxicExposureModal],
+  );
+
+  const hasToxicExposureData = () => {
+    return toxicExposureKeys.some(key => {
+      if (key === 'conditions') return false; // Skip conditions as it's the trigger
+      if (!(key in (data?.toxicExposure || {}))) return false;
+
+      const value = data.toxicExposure[key];
+      switch (typeof value) {
+        case 'boolean':
+          return value === true;
+        case 'string':
+          return value.trim() !== '';
+        case 'number':
+          return true;
+        case 'object':
+          if (value === null) return false;
+          if (Array.isArray(value)) return value.length > 0;
+          return Object.values(value).some(nestedValue => {
+            if (typeof nestedValue === 'boolean') return nestedValue === true;
+            if (typeof nestedValue === 'string')
+              return nestedValue.trim() !== '';
+            return !!nestedValue;
+          });
+        default:
+          return false;
+      }
+    });
+  };
+
+  const deleteToxicExposureData = () => {
+    const deepClone = cloneDeep(data);
+
+    if (deepClone.toxicExposure) {
+      toxicExposureKeys.forEach(key => {
+        if (key !== 'conditions') {
+          delete deepClone.toxicExposure[key];
+        }
+      });
+      // Keep only the 'none' condition selected
+      deepClone.toxicExposure.conditions = { none: true };
+    }
+
+    setFormData(deepClone);
+  };
+
+  // If the user opts to deselect "none" and they are asked in the modal to delete their toxic exposure data,
+  // they can close the modal or click the Cancel button. Doing so should revert the "none" checkbox back to unchecked
+  const revertNoneConditionSelection = () => {
+    const deepClone = cloneDeep(data);
+
+    const toxicExposureSelections = deepClone.toxicExposure?.conditions || {};
+    // Deselect "none" checkbox
+    toxicExposureSelections.none = false;
+
+    if (!deepClone.toxicExposure) {
+      deepClone.toxicExposure = {};
+    }
+    deepClone.toxicExposure.conditions = toxicExposureSelections;
+
+    setFormData(deepClone);
+  };
+
+  const shouldShowDeleteToxicExposureModal = () => {
+    return (
+      data.toxicExposure?.conditions?.none === true && hasToxicExposureData()
```

Comment:
Should selecting 'none' be the only trigger? What about deselecting all conditions, but not selecting 'none'?